### PR TITLE
Coffeelint

### PR DESCRIPTION
**Pull request by kevgo**
@alexdavid @dhakiki 

This adds CoffeeLint. This default configuration is pretty loose, so I tightened it up a bit. This is what I'm going to use on all my Node projects as well.

### CODE REVIEW

File: coffeelint.json [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      "right": 1
+    }
+  },
+
+  "cyclomatic_complexity": {
+    "level": "error"
+  },
+
+  "indentation": {
+    "level": "ignore"     // Disabling because this doesn't handle indented call chains
```
Comment (alexdavid) [inline coffeelint.json]: `.json` doesn't allow comments. We should either make this `.json5` or `.js`/`.coffee` with module.exports line

Comment (kevgo) [inline coffeelint.json]: Fixed by removing the comment

Comment (kevgo) [inline coffeelint.json]: Correction, fixed by renaming to `json5`. Good catch, I didn't know about that file format!

**Comment by alexdavid** [comment]
Thanks for doing this! :ship:

**Pull request closed without merging**